data/src/main/java/pl/edu/agh/papaya/model/Sprint.java

```diff
@@ -12,6 +14,7 @@
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
+import pl.edu.agh.papaya.api.model.SprintState;
```

Comment:
Chociaż moment, stop

Reply:
Ta zależność była przeniesiona wyżej z tego powodu

Reply:
`UserRole` było przy okazji, bo akurat też enum, a ja już sprawdziłam, jak się robi enuma w swaggerze

Reply:
Można ewaluowanie tego jakoś wyjąć do `papaya-web` ~albo można odpuścić i zostawić tę zależność poziom wyżej~

Reply:
no chyba nawet trzeba to przenieść, bo tutaj jest ten sam problem - data zależy od api

Reply:
więc albo wyliczanie SprintState trzeba przenieść do web i korzystać tam z enuma z api, albo zostawić gdzie jest i zwracać enum z modelu + dodać jego odpowiednik w api

Reply:
Eh, no da się zrobiiić. Czym ma być wyliczacz w `web`?

Reply:
~Wspólne~ Niegdyś wspólne utilsy w testach do zduplikowania?

Reply:
a bo ja wiem xd 
Może być metodą prywatną w kontrolerze. Albo random({SprintUtils, SprintHelper, enum SprintState z metodą wyliczającą, [pewnie jeszcze sporo innych]})

Trzeba zrobić to jakoś tak, żeby było najwygodniej korzystać z tego w dwie strony: żeby wyliczyć status dla listy sprintów do zwrócenia, a także żeby wygodnie wyszukać sprinty po dacie (w takim bardziej sqlowym sensie) dla danych statusów (bo nie będzie najwydajniejszym rozwiązaniem pobieranie zawsze wszystkich, wyliczanie statusów i filtrowanie).

Reply:
Ja bym zostawił enuma w Data i enuma w api (nawet mogą być identyczne) z wyliczaniem statusu w model.Sprint, a do tego jakaś klasa (albo metoda w enumie w data), która pozwoli na wyliczenie tych dat (w celach wyszukiwania w bazie danych) po podaniu SprintStatus

Reply:
> Ja bym zostawił enuma w Data i enuma w api (nawet mogą być identyczne) z wyliczaniem statusu w model.Sprint, a do tego jakaś klasa (albo metoda w enumie w data), która pozwoli na wyliczenie tych dat (w celach wyszukiwania w bazie danych) po podaniu SprintStatus

Czyli mamy niepowiązanie niczym identyczne enumy w api i w data, które magicznie zepnie dopiero coś w web?

Reply:
zależy co chcemy mieć - _możemy mieć wszystko, czego tylko zapragniemy!_

Jeśli w data będziemy np. mieć metodę getAllByStatus(SprintStatus status) to ten `SprintStatus` nie może być tym samym, który jest w api. 
Jeśli w data będą tylko metody pobierające po dacie (przedziale,...), to status będzie tylko w api (ale wtedy trzeba też wyliczania w drugą stronę - ze statusu na daty do kryterium wyszukiwania)

Reply:
No wiem, rozumiem, że to są różne rzeczy, ale nie przyszło mi do głowy, że będę dzisiaj kopiować i wklejać enuma stworzonego przez generator jako moją własną klasę, _żeby było ładniej, lepiej i w ogóle_

`Sprint#getSprintStatus()` z tym zduplikowanym enumem mogę zostawić w `data` w takim razie

Jeśli chodzi o _tę drugą stronę_, to umiem sprawić, żeby [coś takiego](https://www.baeldung.com/rest-api-search-language-spring-data-specifications) działało

No i o ile
```
sprintRepository.findAll(new SprintSpecification(LocalDateTime.now(), SprintState.DECLARABLE))
```
wygląda jakoś tak względnie spoko, to w środku to nie jest jakoś nadmiernie piękne